data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaSinkProducer.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.EventHandle;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSinkConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.sink.DLQSink;
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Map;
+
+
+/**
+ * * A helper class which helps takes the buffer data
+ * and produce it to a given kafka topic
+ */
+
+public class KafkaSinkProducer<T> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaSinkProducer.class);
+
+    private final Producer<String, T> producer;
+
+    private final KafkaSinkConfig kafkaSinkConfig;
+
+    private final DLQSink dlqSink;
+
+    private final CachedSchemaRegistryClient schemaRegistryClient;
+
+    private final Collection<EventHandle> bufferedEventHandles;
+
+    private final ExpressionEvaluator expressionEvaluator;
+
+    private final String tagTargetKey;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public KafkaSinkProducer(final Producer producer,
+                             final KafkaSinkConfig kafkaSinkConfig,
+                             final DLQSink dlqSink,
+                             final CachedSchemaRegistryClient schemaRegistryClient,
+                             final ExpressionEvaluator expressionEvaluator,
+                             final String tagTargetKey) {
+        this.producer = producer;
+        this.kafkaSinkConfig = kafkaSinkConfig;
+        this.dlqSink = dlqSink;
+        this.schemaRegistryClient = schemaRegistryClient;
+        bufferedEventHandles = new LinkedList<>();
+        this.expressionEvaluator = expressionEvaluator;
+        this.tagTargetKey = tagTargetKey;
+
+    }
+
+    public void produceRecords(final Record<Event> record) {
+        if (record.getData().getEventHandle() != null) {
+            bufferedEventHandles.add(record.getData().getEventHandle());
+        }
+        TopicConfig topic = kafkaSinkConfig.getTopic();
+        Event event = getEvent(record);
+        final String key = event.formatString(kafkaSinkConfig.getPartitionKey(), expressionEvaluator);
+        Object dataForDlq = event.toJsonString();
+        LOG.info("Producing record " + dataForDlq);
+        try {
+            final String serdeFormat = kafkaSinkConfig.getSerdeFormat();
```

Comment:
this also seem to be a good candidate for class level property

Reply:
we will incorporate this change in next PR.